src/core/backend/pearl/disk_controller.rs

```diff
@@ -0,0 +1,416 @@
+            let vdisk_group = self
+                .groups
+                .read()
+                .await
+                .iter()
+                .find(|g| g.can_process_operation(&op))
+                .cloned();
+            if let Some(group) = vdisk_group {
+                // TODO: check if pearl WorkDirUnavailable error occured and change state
+                group.get(key).await
+            } else {
+                error!("GET[{}] Cannot find storage, operation: {:?}", key, op);
+                Err(Error::vdisk_not_found(op.vdisk_id()))
+            }
+        } else {
+            Err(Error::dc_is_not_available())
+        }
+    }
+
+    pub(crate) async fn get_alien(&self, op: Operation, key: BobKey) -> Result<BobData, Error> {
+        if *self.state.read().await == GroupsState::Ready {
+            let vdisk_group = self.find_group(&op).await;
+            if let Ok(group) = vdisk_group {
+                // TODO: check if pearl WorkDirUnavailable error occured and change state
+                group.get(key).await
+            } else {
+                warn!(
+                    "GET[alien][{}] No alien group has been created for vdisk #{}",
+                    key,
+                    op.vdisk_id()
+                );
+                Err(Error::key_not_found(key))
+            }
+        } else {
+            Err(Error::dc_is_not_available())
+        }
+    }
+
+    pub(crate) async fn exist(
+        &self,
+        operation: Operation,
+        keys: &[BobKey],
+    ) -> Result<Vec<bool>, Error> {
+        if *self.state.read().await == GroupsState::Ready {
+            let group_option = self
+                .groups
+                .read()
+                .await
+                .iter()
+                .find(|g| g.can_process_operation(&operation))
+                .cloned();
+            if let Some(group) = group_option {
+                Ok(group.exist(&keys).await)
+            } else {
+                Err(Error::internal())
+            }
+        } else {
+            Err(Error::dc_is_not_available())
+        }
+    }
+
+    pub(crate) async fn shutdown(&self) {
+        let futures = FuturesUnordered::new();
+        for group in self.groups.read().await.iter() {
+            let holders = group.holders();
+            let holders = holders.read().await;
+            for holder in holders.iter() {
+                let storage = holder.storage().read().await;
+                let storage = storage.storage().clone();
+                let id = holder.get_id();
+                futures.push(async move {
+                    match storage.close().await {
+                        Ok(_) => debug!("holder {} closed", id),
+                        Err(e) => {
+                            error!("error closing holder{}: {} (disk: {:?})", id, e, self.disk)
+                        }
+                    }
+                });
+            }
+        }
+        let _ = futures.collect::<()>().await;
```

Comment:
You assingning `()` unit value

Reply:
This code was written before me (in Pearl strurcture), so I decided to leave it as it is (but seems that there is nothing special in this expression and you confirmed that so I remove it)